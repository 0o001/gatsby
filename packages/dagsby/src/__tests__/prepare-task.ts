const prepareTask = require(`../prepare-task`)
const path = require(`path`)

describe(`prepare-task`, () => {
  it(`prepares a digest of a file in tasks`, async () => {
    let task = {
      func: args => args.a,
      digest: 1,
      files: {
        test: { originPath: path.join(__dirname, `mocks`, `hello.txt`) },
      },
      args: { a: 1 },
    }

    const preparedTask = await prepareTask(task)

    expect(preparedTask).toMatchSnapshot()
    expect(preparedTask.files.test.digest).toBeTruthy()
  })
  it(`prepares a digest of multiple file in tasks`, async () => {
    let task = {
      func: args => args.a,
      digest: 2,
      files: {
        test: { originPath: path.join(__dirname, `mocks`, `hello.txt`) },
        test2: { originPath: path.join(__dirname, `mocks`, `hello2.txt`) },
      },
      args: { a: 1 },
    }

    const preparedTask = await prepareTask(task)

    expect(preparedTask).toMatchSnapshot()
    expect(preparedTask.files.test.digest).toBeTruthy()
    expect(preparedTask.files.test2.digest).toBeTruthy()
  })
})
