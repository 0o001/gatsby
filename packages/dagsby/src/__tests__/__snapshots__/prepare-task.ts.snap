// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prepare-task prepares a digest of a file in tasks 1`] = `
Object {
  "args": Object {
    "a": 1,
  },
  "digest": 1,
  "files": Object {
    "test": Object {
      "digest": "6b8c327f0fc6f470c030a5b6c71154c5",
      "originPath": "<PROJECT_ROOT>/packages/dagsby/src/__tests__/mocks/hello.txt",
    },
  },
  "func": [Function],
}
`;

exports[`prepare-task prepares a digest of multiple file in tasks 1`] = `
Object {
  "args": Object {
    "a": 1,
  },
  "digest": 2,
  "files": Object {
    "test": Object {
      "digest": "6b8c327f0fc6f470c030a5b6c71154c5",
      "originPath": "<PROJECT_ROOT>/packages/dagsby/src/__tests__/mocks/hello.txt",
    },
    "test2": Object {
      "digest": "fa88464ad571637e672fb5bb6d8188e9",
      "originPath": "<PROJECT_ROOT>/packages/dagsby/src/__tests__/mocks/hello2.txt",
    },
  },
  "func": [Function],
}
`;
